### PR TITLE
Command elements postprocessing

### PR DESCRIPTION
**Pull request by ForNeVeR**
It will allow us to properly parse expressions such as `ping 8.8.8.8` or `ping 8.8.ToString` as PowerShell do (slightly hacky, ignoring member access expressions under some conditions, but that's the best I can make).

This pull request will close issue #205.

### CODE REVIEW

**Comment by ForNeVeR** [comment]
I think that's ready. Guys, please review these changes. Personally I don't like all this hackery, but I suspect that's the way PowerShell works.

File: Source/System.Management/Pash/ParserIntrinsics/AstBuilder.cs [hunk 1]
```diff
@@ -2271,6 +2271,31 @@ CommandParameterAst BuildCommandParameterAst(ParseTreeNode parseTreeNode)
                                            new ScriptExtent(parseTreeNode), requiresArgument);
         }
 
+        IEnumerable<CommandElementAst> PostprocessCommandElements(IEnumerable<CommandElementAst> elements)
+        {
+            var queue = new Queue<CommandElementAst>(elements);
+            while (queue.Count > 0)
```
Comment (JayBazuzi) [inline Source/System.Management/Pash/ParserIntrinsics/AstBuilder.cs]: What do you think of replacing `queue.Count > 0` with `queue.Any()`?

Comment (ForNeVeR) [inline Source/System.Management/Pash/ParserIntrinsics/AstBuilder.cs]: I think there's a couple of places where I used `.Count > 0`. I am using it without LINQ for the performance reasons, but if we prefer LINQ way in Pash code, I will replace them.

**Comment by JayBazuzi** [comment]
Yeah, it's unfortunate that this is necessary. Maybe someone will come up with a brilliant way to make this clean and simple. For now, I'll merge it.

File: Source/System.Management/Pash/ParserIntrinsics/AstBuilder.cs (hunk 1 above)
Comment (JayBazuzi) [inline Source/System.Management/Pash/ParserIntrinsics/AstBuilder.cs]: What performance reasons do you have in mind?

While I would write `.Any()`, I deliberately don't have strong coding guidelines in Pash, beyond having tests. So I don't object to `.Count > 0`.

Comment (ForNeVeR) [inline Source/System.Management/Pash/ParserIntrinsics/AstBuilder.cs]: Last time I was checking `.Any()` was about 100 times slower than `.Count > 0`, so I've come up with general rule of using collection-specific methods when they are available.

It is almost never a bottleneck (even being 100 times slower, it's relatively fast operation indeed), but that's the way I used to write my code. I have no strong implication to write it with or without LINQ (so without problem I may change my style to match the project I am working at), but generally I tend to use `.Count` on the collections that have this property.

**Pull request merged**